### ISSUE TITLE
FlutterView.runFromSource lacks error handling

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#13395]
There's a couple of TODOs in this method.
One example of how the problems they refer to might surface is if a hot restart fails for some reason. Today, we just crash the Flutter tool entirely with an ugly stack trace. We don't even report the crash.

cc @johnmccutchan @mit-mit @devoncarew @tvolkert

**Comment by Hixie** [comment]
This would help with hot reload / hot restart trust.

**Comment by tvolkert** [comment]
@Hixie just to be clear, confirming that you're talking about this:

https://github.com/flutter/flutter/blob/adcb5ab4935b6e4c2b71b8f7366626368c7359ee/packages/flutter_tools/lib/src/vmservice.dart#L1296

... and not this:

https://github.com/flutter/engine/blob/5ce89a0f3deeb3b591a30de71351e3ec0120b4c5/shell/platform/android/io/flutter/view/FlutterView.java#L625

**Comment by aam** [comment]
I am not able to repro #13396, do we know of other reasonable ways of forcing restart to fail?

**Comment by csells** [comment]
@Hixie is this issue about the error handling, handling the TODO comments or both?

**Comment by aam** [comment]
I think currently hot restart(FlutterView.runFromSource) errors(like RpcExceptions) are handled by https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/resident_runner.dart#L1225 which exits flutter tools. It looks like this:

````
Launching lib/main.dart on Moto G 4 in debug mode...
Running Gradle task 'assembleDebug'...                                  
Running Gradle task 'assembleDebug'... Done                        11.1s
✓ Built build/app/outputs/apk/debug/app-debug.apk.
Installing build/app/outputs/apk/app.apk...                        11.3s
Syncing files to device Moto G 4...                                     
 6,711ms (!)                                       

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h Repeat this help message.
d Detach (terminate "flutter run" but leave application running).
q Quit (terminate the application on the device).
An Observatory debugger and profiler on Moto G 4 is available at: http://127.0.0.1:33907/Ve-Md1W9jj4=/

Performing hot restart...                                               
Restarted application in 1,407ms.
Application finished.
hot restart failed to complete
```

So this issue got to be more about cleaning up TODOs that error handling.
